plugin-manager/src/main/kotlin/com/itsaky/androidide/plugins/manager/services/IdeCommandServiceImpl.kt

```
@@ -0,0 +1,242 @@
+                    outputChannel.send(CommandOutput.ExitCode(exitCode))
+                    outputChannel.close()
+
+                    val duration = System.currentTimeMillis() - startTime
+                    val result = if (exitCode == 0) {
+                        CommandResult.Success(exitCode, stdoutBuilder.toString(), stderrBuilder.toString(), duration)
+                    } else {
+                        CommandResult.Failure(exitCode, stdoutBuilder.toString(), stderrBuilder.toString(), null, duration)
+                    }
+                    resultDeferred.complete(result)
+                }
+            } catch (e: kotlinx.coroutines.TimeoutCancellationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)